yarn-project/sequencer-client/src/publisher/l1-publisher.ts

```diff
@@ -170,6 +170,9 @@ export class L1Publisher {
   protected account: PrivateKeyAccount;
   protected ethereumSlotDuration: bigint;
 
+  // @note - with blobs, the below estimate seems too large.
+  // Total used for full block from int_l1_pub e2e test: 1m (of which 86k is 1x blob)
+  // Total used for emptier block from above test: 429k (of which 84k is 1x blob)
```

Comment:
@ TMNT team - I didn't change the propose guess from 12m to something closer to reality, as it may be super high to account for the worst case?
Using `integration_l1_publisher.test.ts`, I saw:

- blocks with bloated txs used 1m gas (of which 86k = 1x blob)
- blocks with empty txs used 430k gas (of which 84k = 1x blob)

I also added blob gas estimation below (`blobEvaluationGas`). The contract doesn't waste gas on evaulating empty blobs (if we try and cheat and only evaluate some of 3 used blobs, the header check and epoch proof checks will fail).